### ISSUE TITLE
Rule disallowing passing async functions to subscribe()

### ISSUE THREAD

**Issue by felixfbecker** [cartant/rxjs-tslint-rules#84]
A mistake I often see people less familiar with Rx make is doing async things in `subscribe()`:
```ts
events.subscribe(async value => {
  const data = await fetchData(url, value)
  const moreData = await fetchMoreData(url2, data)
  displayData(moreData)
})
```
This is bad because it does not handle cancellation of the operations and is therefor prone to race conditions.
Instead, it should be this:
```ts
events.pipe(
  switchMap(value => fetchData(url, value)),
  switchMap(value => fetchMoreData(url2, data)),
).subscribe(displayData)
```

This can be done by enforcing that the `subscribe()` observer functions return `void`

**Comment by cartant** [comment]
I've added an `rxjs-no-async-subscribe` rule to 4.22.0.
